crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -296,6 +297,48 @@ where
             status_code: if receipt.success { Some(U64::from(1)) } else { Some(U64::from(0)) },
         };
 
+        // get all receipts for the block
+        let all_receipts = match self.client().receipts_by_block((meta.block_number).into())? {
+            Some(recpts) => recpts,
+            None => return Err(EthApiError::UnknownBlockNumber),
+        };
+
+        // get the previous transaction cumulative gas used
+        let prev_tx_cumulative_gas_used = match all_receipts.get((meta.index - 1) as usize) {
+            Some(prev_receipt) => prev_receipt.cumulative_gas_used,
+            // if meta.index == 0, then there is no previous transaction
+            None => 0,
+        };
```

Comment:
this is currently unsafe, because index can be 0

so we need to check for this